app/livechat/server/hooks/saveAnalyticsData.js

```diff
@@ -12,6 +13,8 @@ callbacks.add('afterSaveMessage', function(message, room) {
 		return message;
 	}
 
+	message = FileUpload.addJWTToFileUrlIfNecessary(message);
```

Comment:
As I commented before, IMO it would be better having a specific helper method to handle this process, and inside that method, I would call the addJWTToFileUrlIfNecessary method.
In that case, we could rename the addJWTToFileUrlIfNecessary to addJWTToFileUrl.